services/venice-router/src/main/java/com/linkedin/venice/router/RouterServer.java

```diff
@@ -639,15 +642,37 @@ public AsyncFuture<LongSupplier> getLongTailRetryMilliseconds(
     if (sslFactory.isPresent()) {
       sslInitializer = new SslInitializer(SslUtils.toAlpiniSSLFactory(sslFactory.get()), false);
       if (config.getClientSslHandshakeThreads() > 0) {
-        ThreadPoolExecutor sslHandshakeExecutor = ThreadPoolFactory.createThreadPool(
-            config.getClientSslHandshakeThreads(),
-            "SSLHandShakeThread",
-            config.getClientSslHandshakeQueueCapacity(),
-            LINKED_BLOCKING_QUEUE);
-        new ThreadPoolStats(metricsRepository, sslHandshakeExecutor, "ssl_handshake_thread_pool");
-        sslInitializer.enableSslTaskExecutor(sslHandshakeExecutor);
+        if (!config.isResolveBeforeSSL()) {
```

Comment:
Can we flip the order of `if`s? `if (config.isResolveBeforeSSL()) {...} else {...}`

Reply:
Yeah, that's easier to read. Updated!